setup_ib.c

```diff
@@ -31,7 +31,13 @@ int connect_qp_server ()
     /* init local qp_info */
     local_qp_info.lid	 = ib_res.port_attr.lid; 
     local_qp_info.qp_num = ib_res.qp->qp_num;
+    local_qp_info.gid_index = ib_res.gid_index;
     
+    if (local_qp_info.lid == 0 && ib_res.port_attr.link_layer == IBV_LINK_LAYER_ETHERNET) {
+        ret = ibv_query_gid(ib_res.ctx, IB_PORT, local_qp_info.gid_index, &local_qp_info.gid);
```

Comment:
A suggested fix for LINE#37:

ret = ibv_query_gid(ib_res.ctx, config_info.gid_index, config_info.dev_index, &local_qp_info.gid);